### PR TITLE
Add block for old PowerShell Core. Add test for old PowerShell Core.

### PR DESCRIPTION
**Pull request by brywang-msft**
Block older versions of PowerShell Core that will fail to load PackageManagement.

Also adds a script to test import-module on older PowerShell Core.

### CODE REVIEW

File: src/Microsoft.PowerShell.PackageManagement/PackageManagement.psm1 [hunk 1]
```diff
@@ -2,8 +2,22 @@
 # Script module for module 'PackageManagement'
 #
 Set-StrictMode -Version Latest
+Microsoft.PowerShell.Utility\Import-LocalizedData  LocalizedData -filename PackageManagement.Resources.psd1
 
 # Summary: PackageManagement is supported on Windows PowerShell 3.0 or later, Nano Server and PowerShellCore
+$isCore = ($PSVersionTable.Keys -contains "PSEdition") -and ($PSVersionTable.PSEdition -ne 'Desktop')
+Write-Verbose "Core: $isCore"
```
Comment (brywang-msft) [inline src/Microsoft.PowerShell.PackageManagement/PackageManagement.psm1]: Oops, should remove this debug call

File: src/Microsoft.PowerShell.PackageManagement/PackageManagement.Resources.psd1 [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# Localized PackageManagement.Resources.psd1
+#
+#########################################################################################
+
+ConvertFrom-StringData @'
+###PSLOC
+
+    OldPowerShellCoreVersion=PowerShell Core {0} is no longer supported by Package Management. Please update to the most recent version of PowerShell Core to get the latest features and bug fixes.
```
Comment (jianyunt) [inline src/Microsoft.PowerShell.PackageManagement/PackageManagement.Resources.psd1]: Suggest:  

   PackageManagement is no longer support PowerShell Core '{0}'. Please install the latest PowerShell Core and try again.

File: Test/PSL/win10/PSL_6.0.0.14.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+	"PowerShell": {
+		"displayName": "PowerShell_6.0.0.14",
+		"source": "https://github.com/PowerShell/PowerShell/releases/download/v6.0.0-alpha.14/PowerShell_6.0.0.14-alpha.14-win10-x64.msi",
```
Comment (jianyunt) [inline Test/PSL/win10/PSL_6.0.0.14.json]: >.0.0-alpha.14 [](start = 74, length = 13)

what's the reason we choose this particular version?

File: src/Microsoft.PowerShell.PackageManagement/PackageManagement.psm1 [hunk 4]
```diff
@@ -2,8 +2,22 @@
...
+if ($isCore)
+{
+    # PackageManagement only versions of PowerShell Core built on .NETCoreApp 2.0 and above
```
Comment (brywang-msft) [inline src/Microsoft.PowerShell.PackageManagement/PackageManagement.psm1]: *only works

File: Test/run-tests.ps1 [hunk 5]
```diff
@@ -465,6 +460,15 @@ if ($testframework -eq "coreclr")
     }
 }
 
+Write-Host "Running OOB tests"
```
Comment (brywang-msft) [inline Test/run-tests.ps1]: Move to Pester:

- [ ] Download old PSCore version as part of test setup
- [ ] Add 'Legacy' tag
- [ ] Invoke Pester for 'Legacy' tests only on old PSCore

File: src/Microsoft.PowerShell.PackageManagement/PackageManagement.Resources.psd1 (hunk 2 above)
Comment (brywang-msft) [inline src/Microsoft.PowerShell.PackageManagement/PackageManagement.Resources.psd1]: @JKeithB, are you the right person to review new resources?

File: Test/PSL/win10/PSL_6.0.0.14.json (hunk 3 above)
Comment (brywang-msft) [inline Test/PSL/win10/PSL_6.0.0.14.json]: Just that it's an older version of PSCore that isn't built on netcoreapp2.0. Any version that meets that requirement is equivalent.

**Comment by brywang-msft** [comment]
Can we link to the GitHub release page with a fwlink? Might be simpler.

**Pull request closed without merging**